editors/code/src/spawn.ts

```diff
@@ -0,0 +1,133 @@
+import * as vscode from "vscode";
+
+import {
+  ChildProcessWithoutNullStreams,
+  SpawnOptionsWithoutStdio,
+  spawn,
+} from "child_process";
+
+type ChildProcessRejection = {
+  interactive: boolean;
+  code: number;
+  reason: string;
+};
+
+const canSpawnInteractive = (platform: NodeJS.Platform): boolean => {
+  switch (platform) {
+    case "darwin":
+      return true;
+    case "linux":
+      return true;
+    default:
+      return false;
+  }
+};
+
+const handleChildProcessPromise = (
+  cp: ChildProcessWithoutNullStreams,
+  resolve: (value: string) => void,
+  reject: (reason: ChildProcessRejection) => void,
+  interactive: boolean
+) => {
+  // Set stdout and stderr encoding + pipe to plain strings
+  let stdout = "";
+  let stderr = "";
+  cp.stdout.setEncoding("utf8");
+  cp.stderr.setEncoding("utf8");
+  cp.stdout.on("data", (data) => (stdout += data));
+  cp.stderr.on("data", (data) => (stderr += data));
+  // Reject if spawning errors out
+  cp.on("error", (e) => {
+    reject({
+      interactive,
+      code: -1,
+      reason: e.message,
+    });
+  });
+  // Resolve with stdout or reject with stderr
+  // depending on the exist code and/or data
+  cp.on("close", (code) => {
+    if (code === 0 && stderr.length === 0) {
+      resolve(stdout);
+    } else {
+      reject({
+        interactive,
+        code: code || -1,
+        reason: stderr,
+      });
+    }
+  });
+};
+
+const spawnNaive = (
+  command: string,
+  args: string[],
+  options?: SpawnOptionsWithoutStdio
+): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    // Spawn the wanted command directly
+    const cp = spawn(command, args, {
+      cwd: process.cwd(),
```

Comment:
cwd should use the one passed in options if available
```suggestion
      cwd: options?.cwd ?? process.cwd(),
```

Reply:
`cwd` should be overwritten by any options passed in with the `...options` on line `72`

Reply:
Ah yeah, makes sense, my bad